MohammadBallal/app/routes.py

```diff
@@ -0,0 +1,98 @@
+from fastapi import APIRouter, HTTPException, Path
+from app.models import Task, TaskModel, UUID, TaskUpdate
+
+router = APIRouter()
+
+tasks = []
+urgent_tasks = []
+
+
+# Create a new task
+@router.post("/tasks/")
+def create_task(task_data: TaskModel):
+    task = Task(**task_data.model_dump())
+    if task.priority:
+        urgent_tasks.append(task)
+        return {"message": "Urgent task created", "task": task.task_dict()}
+    tasks.append(task)
+    return {"message": "Normal task created", "task": task.task_dict()}
+
+
+# Get All Tasks
+@router.get("/tasks/")
+def get_all_tasks():
+    all_tasks = tasks + urgent_tasks
+    return {"tasks": [task.task_dict() for task in all_tasks]}
+
+
+# Get Urgent Tasks
+@router.get("/tasks/urgent/")
+def get_urgent_tasks():
+    if not urgent_tasks:
+        return {"message": "No urgent tasks found."}
+    return {"urgent_tasks": [task.task_dict() for task in urgent_tasks]}
+
+
+# Get Task by ID
+@router.get("/tasks/{task_id}/")
+def get_task_by_id(task_id: UUID = Path(..., title="The ID of the task to get")):
+    for task_list in (urgent_tasks, tasks):
+        for task in task_list:
+            if task.task_id == task_id:
+                return {"task": task.task_dict()}
+    raise HTTPException(status_code=404, detail="Task not found")
+
+
+# Update a Task by ID
+@router.put("/tasks/{task_id}/")
+def update_task(
+    *,
+    task_id: UUID = Path(..., title="The ID of the task to update"),
+    update_data: TaskUpdate
+):
+    # Find the task to update
+    task_to_update = next((task for task in tasks + urgent_tasks if task.task_id == task_id), None)
+    if not task_to_update:
+        raise HTTPException(status_code=404, detail="Task not found")
+
+    # Remove the task from its current list
+    if task_to_update in tasks:
+        tasks.remove(task_to_update)
+    elif task_to_update in urgent_tasks:
+        urgent_tasks.remove(task_to_update)
+
+    # Update task fields
+    if update_data.title is not None:
```

Comment:
What happens if the attributes are more? 10? 20?
are u sure to use this way?
let us discuss the problem: you do the same operations for all properties for an object, I think it is better to use a loop to iterate over all of them and handle these operations.

Reply:
You’re right. I treated the attributes as if they were limited, but as you mentioned it’s better to use a loop when there are many attributes. It's done.